### PR TITLE
add aggregated cluster roles for crds

### PR DESCRIPTION
**Pull request by rinckm**
### Description

Add aggregate cluster roles for the view, edit and admin role for tenants and pipelineruns.

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [ ] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [x] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: charts/steward/templates/clusterrole-aggregate.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - update
+      - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: steward-aggregate-view
+  labels:
+    {{- include "steward.labels" . | nindent 4 }}
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
```
Comment (anbrsap) [inline charts/steward/templates/clusterrole-aggregate.yaml:31]: Shouldn't it be aggregated to edit and admin, too. Then we can leave out these permissions in `steward-aggregate-edit`.

File: changelog.yaml [hunk 2]
```diff
@@ -52,6 +52,14 @@
   date: TBD
   changes:
 
+    - type: enhancement
+      impact: minor
+      title: Add aggregated cluster roles for crds
+      description: |-
+        Add aggregated cluster roles for the generic view edit and admin roles 
+        for the custom resources tenants and pipelines.
+      pullRequestNumber: 0
```
Comment (rinckm) [inline changelog.yaml:61]:
```suggestion
      pullRequestNumber: 340
```

File: changelog.yaml [hunk 3]
```diff
@@ -52,6 +52,14 @@
   date: TBD
   changes:
 
+    - type: enhancement
+      impact: minor
+      title: Add aggregated cluster roles for crds
+      description: |-
+        Add aggregated cluster roles for the generic view edit and admin roles 
+        for the custom resources tenants and pipelines.
```
Comment (alxsap) [inline changelog.yaml:60]:
```suggestion
        for the custom resources tenants and pipelineruns.
```

File: charts/steward/templates/clusterroles-crds.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: steward-edit
+  labels:
+     {{- include "steward.labels" . | nindent 4 }}
```
Comment (alxsap) [inline charts/steward/templates/clusterroles-crds.yaml:6]:
```suggestion
    {{- include "steward.labels" . | nindent 4 }}

```

**Review by alxsap — APPROVED** [review]

**Pull request merged**